pkg/upgradeservice/bootstrap.go

```diff
@@ -40,6 +46,41 @@ func bootstrap(params types.UpgradeServiceParams) (finalError error) {
 	return nil
 }
 
+// updateWithinKubeRange checks if the update version is within the same major version and
+// at most one minor version ahead of the current version.
+func updateWithinKubeRange(params types.UpgradeServiceParams) error {
+	currentVersion, err := extractKubeVersion(params.CurrentECVersion)
+	if err != nil {
+		return errors.Wrap(err, "failed to extract current kube version")
+	}
+	updateVersion, err := extractKubeVersion(params.UpdateECVersion)
+	if err != nil {
+		return errors.Wrap(err, "failed to extract update kube version")
+	}
+	if currentVersion.Major() != updateVersion.Major() {
+		return errors.Errorf("major version mismatch: current %s, update %s", currentVersion.String(), updateVersion.String())
```

Comment:
the %s formatter calls the string function when implemented

https://pkg.go.dev/fmt#Stringer

```suggestion
		return errors.Errorf("major version mismatch: current %s, update %s", currentVersion, updateVersion)
```